google-cloud-tools-plugin/testSrc/com/intellij/appengine/AppEngineCodeInsightTestCase.java

```diff
@@ -61,6 +62,8 @@
   @Override
   protected void setUp() throws Exception {
     super.setUp();
+    // Fixes https://youtrack.jetbrains.com/issue/IDEA-129297. Only occurs in Jenkins.
+    VfsRootAccess.allowRootAccess(System.getProperty("user.dir"));
```

Comment:
are the invokeAndWait changes and this change related?

Reply:
There are 2 things going on here:

1) `AppEngineInsightTestCase` was failing with that root access error - adding this `allowRootAccess` fixes the problem
2) I refactored `DefaultAppEngineProjectServiceTest` to use the new CloudToolsRule. This allowed me to 1) get rid of the `allowRootAccess` fix, and 2) forced me to wrap the read actions in an invokeAndWait so they are run on the EDT.

The reasons behind #2 are still unclear to me. I'm still investigating. I think I may roll back part 2 just to keep this PR concise.